robotoff/app/core.py

```diff
@@ -125,13 +125,39 @@ def get_insights(
 
         elif order_by == "n_votes":
             query = query.order_by(ProductInsight.n_votes.desc())
-
     if as_dict:
         query = query.dicts()
 
     return query.iterator()
 
 
+def get_predictions(
+    barcode: Optional[str] = None,
+    value_tag: Optional[str] = None,
+    server_domain: Optional[str] = None,
+    limit: Optional[int] = 25,
+    offset: Optional[int] = None,
+    count: bool = False,
+) -> Iterable[Prediction]:
+    if server_domain is None:
+        server_domain = settings.OFF_SERVER_DOMAIN
+
+    where_clauses = [Prediction.server_domain == server_domain]
+
+    if barcode:
+        where_clauses.append(Prediction.barcode == barcode)
+
+    if value_tag:
+        where_clauses.append(Prediction.value_tag == value_tag)
+
+    query = Prediction.select()
+
+    if where_clauses:
+        query = query.where(*where_clauses)
+
+    return query.iterator()
```

Comment:
depending on the "count" parameter, you should not return this, look at get_insights to see how they do it.
```suggestion
    if count:
         return query.count()
    else:
        return query.iterator()
```